server/src/main/java/org/jboss/as/server/services/net/LocalDestinationOutboundSocketBindingAddHandler.java

```diff
@@ -67,6 +74,51 @@ protected boolean requiresRuntime(OperationContext context) {
     }
 
     @Override
+    protected void populateModel(final OperationContext context, final ModelNode operation, final Resource resource)
+            throws OperationFailedException {
+        final PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
+        final String socketBindingGroupName = address.getParent().getLastElement().getValue();
+        final String outboundSocketBindingName = address.getLastElement().getValue();
+
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+
+                Resource resource;
+
+                if (!context.getProcessType().isServer()) {
+                    resource = context.readResourceFromRoot(context.getCurrentAddress().getParent(), false);
+                }else{
+                    resource = context.readResourceFromRoot(PathAddress.pathAddress(ModelDescriptionConstants.SOCKET_BINDING_GROUP, socketBindingGroupName), false);
+                }
+
+                Set<String> socketBindingNames = resource.getChildrenNames(ModelDescriptionConstants.SOCKET_BINDING);
+                for (String name : socketBindingNames) {
+                    if (name.equalsIgnoreCase(outboundSocketBindingName)) {
+                        throwEx(name, socketBindingGroupName);
+                    }
+                }
```

Comment:
I don't think there's any requirement to be unique ignoring case, and all of these can just be a test of "if (existingnames.contains(newname)". If somehow there's some code that fails in the case of equalsIgnoreCase that code is likely buggy. These strings are used in resource addresses, MSC service names, and capabilty names and none of those require uniqueness ignoring case.